Published/asc052.md

```diff
@@ -0,0 +1,49 @@
+#### **Title**
+Modifying instruction: `liquid_handle`
+
+#### **Authorship**
+Asuka Ota <asuka.ota@strateos.com>
+
+#### **Motivation**
+`liquid handle` allows the transfer of liquid from one container to one or more other containers. The viscosity of
+liquid samples used in the laboratory ranges from very fluid (ie. acetone) to very thick (ie. glycerol). A limitation
+to the current `liquid_handle` instruction is that there are only two modes of volumetric manipulation available:
+dispense and air-displacement. While this is sufficient for liquids with relatively low viscosity, they should not be
+used for viscous samples due to the inability for air pressure to overcome surface tension. To provide a more accurate
+way to handle these samples, 'positive-displacement' method is added to `mode`, and 'viscous' is added to
+`liquid_class`. Furthermore, there are devices that are capable of requesting liquid transfer amount in mass instead of volume. To allow such , optional density parameter
+should be added to convert mass to volume (and vice versa) as needed.
+
+#### **Proposal**
+We propose a more versatile `liquid_handle` instruction to support positive-displacement mode, and optional density
+field in addition to the currently-available capabilities.
+
+```
+{
+  “op”: “liquid_handle”,
+  /* Parameters describing the transport of liquid locations, speed, acceleration and any other device-specific parameters that may need to be provided by the user. Optional */
+  “locations”: List[{
+    transports: [
+        {
+          /* Density of the liquid being handled. If this value is specified, the transport volume is converted
+          to mass. Optional. */
+          "density": Unit
+          "mode_params": [{
+            /* Type of liquid transported. Optional. */
+            "liquid_class": Enum("Air", "Default", "Protein", "Viscous")
+          }]
+          ...
+        }
+    ],
+    ...
+  }],
+  /* Allowed modes for liquid handling. 'positive-displacement' should be included as a new mode. 'positive-displacement' mode will use the same mode_params as other modes. */
+  “mode”: Enum(“air-displacement”, ”dispense”, “positive-displacement”)
```

Comment:
hmm there is a discrepancy here. Seems like its `air_displacement` in the currnet specification. Let's verify it it should be snake_case or kebab-case

Reply:
good catch! it's supposed to be snake. thanks!